### PR TITLE
Update requirements and fix issues from upgrade

### LINKED ISSUE #334: TypeError: Cannot compare tz-naive and tz-aware datetime-like objects.
Sentry Issue: [API-2F](https://sentry.io/organizations/solararbiter-rx/issues/1498639135/?referrer=github_integration)

```
KeyError: 1581058800000000000
  File "pandas/_libs/index.pyx", line 463, in pandas._libs.index.DatetimeEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 997, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1004, in pandas._libs.hashtable.Int64HashTable.get_item

KeyError: Timestamp('2020-02-07 07:00:00')
  File "pandas/core/indexes/base.py", line 2646, in get_loc
    return self._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 433, in pandas._libs.index.DatetimeEngine.get_loc
  File "pandas/_libs/index.pyx", line 465, in pandas._libs.index.DatetimeEngine.get_loc

KeyError: 1581058800000000000
  File "pandas/_libs/index.pyx", line 463, in pandas._libs.index.DatetimeEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 997, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1004, in pandas._libs.hashtable.Int64HashTable.get_item

KeyError: Timestamp('2020-02-07 07:00:00')
  File "pandas/core/indexes/base.py", line 4842, in get_slice_bound
    slc = self.get_loc(label)
  File "pandas/core/indexes/datetimes.py", line 699, in get_loc
    return Index.get_loc(self, key, method, tolerance)
  File "pandas/core/indexes/base.py", line 2648, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 433, in pandas._libs.index.DatetimeEngine.get_loc
  File "pandas/_libs/index.pyx", line 465, in pandas._libs.index.DatetimeEngine.get_loc

TypeError: Cannot compare tz-naive and tz-aware datetime-like objects.
(17 additional frame(s) were not displayed)
...
  File "pandas/core/indexes/base.py", line 4845, in get_slice_bound
    return self._searchsorted_monotonic(label, side)
  File "pandas/core/indexes/base.py", line 4796, in _searchsorted_monotonic
    return self.searchsorted(label, side=side)
  File "pandas/core/indexes/datetimes.py", line 855, in searchsorted
    self._data._check_compatible_with(value)
  File "pandas/core/arrays/datetimes.py", line 462, in _check_compatible_with
    self._assert_tzawareness_compat(other)
  File "pandas/core/arrays/datetimes.py", line 643, in _assert_tzawareness_compat
    "Cannot compare tz-naive and tz-aware datetime-like objects."
```

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #334 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Includes lots of changes to resolve deprecation warnings

### CODE REVIEW

**Comment by wholmgren** [comment]
Are you going to revert the >= to == when you’re happy with a new version set?

**Comment by alorenzo175** [comment]
I'm not sure. The dashboard and API do not have fixed requirements so what happens is that core is mostly used with the newest  versions. We could lock those down too. One the other hand, it's probably useful to test on the latest versions but we should probably use the versions it's tested against. Do you have an opinion?

**Comment by alorenzo175** [comment]
maybe the answer is to pin in requirements.txt for everything, then test against that and the updated versions

**Comment by wholmgren** [comment]
My understanding is that unpinned requirements are a reliability and most importantly a security risk in production. I'd be fine with unpinned requirements in a dev environment if a test environment uses the same configuration as production.

**Comment by alorenzo175** [comment]
Ok, I've fixed the requirements and now each package will be tested with whatever versions are installed before being deployed thanks to https://github.com/SolarArbiter/openshift-deployment/pull/35/files

File: solarforecastarbiter/io/nwp.py [hunk 1]
```diff
@@ -104,6 +104,10 @@ def load_forecast(
     mapping_subset = {k: v for k, v in CF_MAPPING.items() if v in variables}
 
     limit = 500  # maximum distance from point to closest grid point
+    if start.tzinfo is not None:
```
Comment (wholmgren) [inline solarforecastarbiter/io/nwp.py]: why do we do this? comment might help maintainability

**Pull request merged**